.github/workflows/pr.yml

```diff
@@ -0,0 +1,61 @@
+name: UI5-Language-Assistant
+
+on:
+  
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    env:
+      HUSKY_SKIP: "true"
+      # NODE_OPTIONS: "--openssl-legacy-provider"
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+        os: [macos-latest, ubuntu-latest, windows-latest]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'yarn'
+    - name: Setup yarn
```

Comment:
afaik `setup-node` action supports yarn so no need for a separate yarn install step
- https://github.com/actions/setup-node